cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -207,9 +207,7 @@ func add(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 		if delErr != nil {
 			log.Errorf("Error received from DelNetwork grpc call for container %s: %v",
 				args.ContainerID, delErr)
-		}
-
-		if !r.Success {
+		} else if !r.Success {
```

Comment:
nit: We will end up here only if delErr is nil right? So, don't see any value printing delErr in the below error message.

Reply:
Also, why not extend this to scenarios where add fails as well.

https://github.com/aws/amazon-vpc-cni-k8s/blob/master/cmd/routed-eni-cni-plugin/cni.go#L164

Reply:
@achevuru because when add fails, we always return within `err != nil `check.

Reply:
> nit: We will end up here only if delErr is nil right? So, don't see any value printing delErr in the below error message.

make sense.

Reply:
I think later sometime we need to simplify this, like for instance make Add and Del functions return just one structure with the response and error, now there are two variables and anyone adding a piece of code should be aware of setting response when err is nil.

Reply:
yeah, this whole piece needs to restructured and better handled.
like how to handle err vs r.success. (seems there is a duplicate here)

Reply:
yeah agreed.